### PR TITLE
Removed `PromotionalOffer.IntroDurationType` in favor of `PaymentMode`

### PR DESCRIPTION
**Pull request by NachoSoto**
See #1045.
Depends on #1107.

### CODE REVIEW

File: PurchasesTests/Purchasing/ProductInfoTests.swift [hunk 1]
```diff
@@ -150,6 +150,6 @@ class ProductInfoTests: XCTestCase {
                                                introPrice: 0,
                                                subscriptionGroup: "cool_group",
                                                discounts: [discount1, discount2, discount3])
-        expect(productInfo.cacheKey) == "cool_product-49.99-UYU-1-0-cool_group-P3Y-P3W-0-offerid1-offerid2-offerid3"
+        expect(productInfo.cacheKey) == "cool_product-49.99-UYU-1-0-cool_group-P3Y-P3W-2-offerid1-offerid2-offerid3"
```
Comment (NachoSoto) [inline PurchasesTests/Purchasing/ProductInfoTests.swift:153]: This doesn't need to be stable across versions.

File: Purchases/Purchasing/ProductInfo.swift [hunk 2]
```diff
@@ -24,7 +24,7 @@ struct ProductInfo {
     let price: Decimal
     let normalDuration: String?
     let introDuration: String?
-    let introDurationType: PromotionalOffer.IntroDurationType
+    let introDurationType: PromotionalOffer.PaymentMode
```
Comment (NachoSoto) [inline Purchases/Purchasing/ProductInfo.swift:27]: This value isn't actually encoded in `asDictionary`, so the fact that its `rawValue`s are slightly different doesn't have an effect on the API requests.

File: Purchases/Purchasing/PromotionalOffer.swift [hunk 3]
```diff
@@ -28,15 +28,6 @@ public class PromotionalOffer: NSObject {
 
     }
 
-    // Fixme: remove in favor of `PaymentMode`: https://github.com/RevenueCat/purchases-ios/issues/1045
-    internal enum IntroDurationType: Int {
```
Comment (taquitos) [inline Purchases/Purchasing/PromotionalOffer.swift:32]: bye-bye-bye-bye

File: PurchasesTests/Purchasing/ProductInfoTests.swift (hunk 1 above)
Comment (taquitos) [reply PurchasesTests/Purchasing/ProductInfoTests.swift:153]: Only slightly matters to cached product infos, but no real harm.

**Review by taquitos — APPROVED** [review]
👋 🐐

File: PurchasesTests/Purchasing/ProductInfoTests.swift (hunk 1 above)
Comment (NachoSoto) [reply PurchasesTests/Purchasing/ProductInfoTests.swift:184]: Are they cached on disk though?

**Pull request merged**